### ISSUE TITLE
Context graph

### ISSUE THREAD

**Issue by ezerhouni** [k2-fsa/icefall#644]
I was wondering if there is any plan on adding context graph to k2/Icefall (something similar to :https://wenet.org.cn/wenet/context.html )

cc @csukuangfj

**Comment by marcoyang1998** [comment]
Yes, it is one of our plans and we‘ve made initial progress on this.

**Comment by ezerhouni** [comment]
@marcoyang1998 Thanks ! Do you need a hand ? I started working also on it

**Comment by csukuangfj** [comment]
@ezerhouni 

could you have a look at https://github.com/k2-fsa/icefall/pull/645 and help propagate the changes to other folders in librispeech?

**Comment by ezerhouni** [comment]
@csukuangfj Let me have a look. I can start doing the propagation tomorrow. 
In the coming days I will open a PR for context graph, and if you have time to have a look it would be great

**Comment by csukuangfj** [comment]
> @csukuangfj Let me have a look. I can start doing the propagation tomorrow. 
> 
> In the coming days I will open a PR for context graph, and if you have time to have a look it would be great

Sure, thanks!

**Comment by ezerhouni** [comment]
@csukuangfj Also, in the last implementation ```modified_beam_search_ngram_rescoring```, I might have found an error
